doc/rules/control_structure/single_expression_per_line.rst

```diff
@@ -0,0 +1,166 @@
+===================================
+Rule ``single_expression_per_line``
+===================================
+
+Multi-line arrays, arguments list, parameters list, control structures,
+``switch`` cases and ``match`` expressions should have one element by line.
+
+Configuration
+-------------
+
+``elements``
+~~~~~~~~~~~~
+
+Which expression must have one element by line.
+
+Allowed values: a subset of ``['arguments', 'arrays', 'case', 'control_structures', 'match', 'parameters']``
+
+Default value: ``['arrays']``
+
+Examples
+--------
+
+Example #1
+~~~~~~~~~~
+
+*Default* configuration.
+
+.. code-block:: diff
+
+   --- Original
+   +++ New
+    <?php
+   -array(1,
+   -    2);
+   +array(
+   +1,
+   +    2
+   +);
+
+Example #2
+~~~~~~~~~~
+
+With configuration: ``['elements' => ['arguments']]``.
+
+.. code-block:: diff
+
+   --- Original
+   +++ New
+    <?php
+   -foo(1,
+   -    2);
+   +foo(
+   +1,
+   +    2
+   +);
+
+Example #3
+~~~~~~~~~~
+
+With configuration: ``['elements' => ['control_structures']]``.
+
+.. code-block:: diff
+
+   --- Original
+   +++ New
+    <?php
+   -if ($a
+   -    && $b) {};
+   +if (
+   +$a
+   +    && $b
+   +) {};
+
+Example #4
+~~~~~~~~~~
+
+With configuration: ``['elements' => ['case']]``.
+
+.. code-block:: diff
+
+   --- Original
+   +++ New
+    <?php
+    switch ($foo) {
+   -    case 0: case 1:
+   +    case 0:
+   +case 1:
+            return null;
+        };
+
+Example #5
+~~~~~~~~~~
+
+With configuration: ``['elements' => ['parameters']]``.
+
+.. code-block:: diff
+
+   --- Original
+   +++ New
+    <?php
+   -function foo($x,
+   -    $y)
+   +function foo(
+   +$x,
+   +    $y
+   +)
+    {
+    }
+
+Example #6
+~~~~~~~~~~
+
+With configuration: ``['elements' => ['match']]``.
+
+.. code-block:: diff
+
+   --- Original
+   +++ New
+    <?php
+    match($x) {
+   -    1 => 1, 2 => 2
+   +    1 => 1,
+   +2 => 2
+    };
+
+Rule sets
+---------
+
+The rule is part of the following rule sets:
+
+- `@PER <./../../ruleSets/PER.rst>`_ with config:
+
+  ``['array', 'arguments', 'parameters', 'control_structures', 'case', 'match']``
+
+- `@PER-CS <./../../ruleSets/PER-CS.rst>`_ with config:
+
+  ``['array', 'arguments', 'parameters', 'control_structures', 'case', 'match']``
+
+- `@PER-CS1.0 <./../../ruleSets/PER-CS1.0.rst>`_ with config:
+
+  ``['array', 'arguments', 'parameters', 'control_structures', 'case', 'match']``
+
+- `@PER-CS2.0 <./../../ruleSets/PER-CS2.0.rst>`_ with config:
+
+  ``['array', 'arguments', 'parameters', 'control_structures', 'case', 'match']``
+
+- `@PSR12 <./../../ruleSets/PSR12.rst>`_ with config:
+
+  ``['array', 'arguments', 'parameters', 'control_structures', 'case', 'match']``
+
+- `@PhpCsFixer <./../../ruleSets/PhpCsFixer.rst>`_ with config:
+
+  ``['array', 'arguments', 'parameters', 'control_structures', 'case', 'match']``
+
+- `@Symfony <./../../ruleSets/Symfony.rst>`_ with config:
+
+  ``['array', 'arguments', 'parameters', 'control_structures', 'case', 'match']``
```

Comment:
When looking at PSR12 (https://www.php-fig.org/psr/psr-12/) I think the right config is to enable the fixer on everything.

But maybe `@PhpCsFixer` and `@Symfony` rule set will prefer to override the PSR12 definition to match there existing code base.